### PR TITLE
Pic 4199 - Add IsYouth to the CHER

### PR DESCRIPTION
**Pull request by johnyfarrar**

### CODE REVIEW

**Review by jpanesar07 — DISMISSED** [review]

**Comment by markreesmoj** [comment]
It would be great to see an integration test for this to prove that `isYouth` gets added to the message which is put onto the topic

Maybe expanding on this?

https://github.com/ministryofjustice/court-hearing-event-receiver/blob/c3a83a7f025babe2c005b95b0da06cd883fde065/src/test/kotlin/uk/gov/justice/digital/hmpps/courthearingeventreceiver/controller/EventControllerIntTest.kt#L57

**Review by markreesmoj — APPROVED** [review]
👍

File: src/main/kotlin/uk/gov/justice/digital/hmpps/courthearingeventreceiver/model/Defendant.kt [hunk 1]
```diff
@@ -40,4 +40,7 @@ data class Defendant(
 
   @JsonProperty("croNumber")
   val croNumber: String?,
+
+  @JsonProperty("isYouth")
+  val isYouth: Boolean,
```
Comment (markreesmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/courthearingeventreceiver/model/Defendant.kt:45]: It is probably not too important, but using `Boolean` as opposed to `boolean` introduces the possibility that the value might be `null` 
`boolean` will default it to false if it is not present.

Comment (markreesmoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/courthearingeventreceiver/model/Defendant.kt:45]: If it is always present obviously this does not matter. I am not certain how Jackson handles nulls downstream, might be worth an Integration test for this to make sure we understand the behaviour. Or alternatively just going with the primitive `boolean`

Comment (johnyfarrar) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/courthearingeventreceiver/model/Defendant.kt]: Updated to primitive `boolean`

**Review by markreesmoj — APPROVED** [review]
👍

File: src/main/kotlin/uk/gov/justice/digital/hmpps/courthearingeventreceiver/model/Defendant.kt (hunk 1 above)
Comment (markreesmoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/courthearingeventreceiver/model/Defendant.kt]: Ignore this, totally forgot this is a Kotlin project where the tri-state Boolean has been abolished - see https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-boolean/

Comment (markreesmoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/courthearingeventreceiver/model/Defendant.kt]: Sorry

**Pull request merged**